File: src/java/com/codecool/gameoflife/view/GridPaneView.java
package com.codecool.gameoflife.view;

import com.codecool.gameoflife.controller.GameController;
import com.codecool.gameoflife.model.Cell;
import javafx.scene.layout.GridPane;

public class GridPaneView extends GridPane {

    private CellView[][] cells;
    private boolean gridOn;


    public GridPaneView(GameController controller) {
        Cell[][] gameBoard = controller.getGameBoard();
        this.gridOn = false;
        this.cells = new CellView[gameBoard.length][gameBoard[0].length];
        this.createBoardView(gameBoard, controller);
    }

    public void createBoardView(Cell[][] gameBoard, GameController controller) {
        for (int i = 0; i < gameBoard.length; i++) {
            for (int j = 0; j < gameBoard[i].length; j++) {
                cells[i][j] = new CellView(gameBoard[i][j]);
                int x = i;
                int y = j;
                cells[i][j].setOnMousePressed(event -> controller.updateCell(x, y));
                this.add(cells[i][j], j, i);
            }
        }
    }

    public void updateCells(Cell[][] gameBoard) {
        for (int i = 0; i < gameBoard.length; i++) {
            for (int j = 0; j < gameBoard[i].length; j++) {
                cells[i][j].update(gameBoard[i][j], gridOn);
            }
        }
    }

    public void setGridOn(boolean visible) {
        this.gridOn = visible;
    }

    public void updateSingleCell(Cell cellToUpdate, int x, int y) {
        this.cells[x][y].update(cellToUpdate, gridOn);
    }
}

File: src/kotlin/com/codecool/gameoflife/view/GridPaneView.kt
package com.codecool.gameoflife.view

import com.codecool.gameoflife.controller.GameController
import javafx.event.EventHandler
import javafx.scene.layout.GridPane

class GridPaneView(private val controller: GameController) : GridPane() {

    private val cells: Array<Array<CellView>>
    var gridOn: Boolean = false

    init {
        val xSize = controller.gameBoard.width
        val ySize = controller.gameBoard.height
        gridOn = false
        cells = Array(xSize) { i -> Array(ySize) { j ->
            val c = CellView(controller.gameBoard.board[i][j]).apply {
                onMousePressed = EventHandler { controller.updateCell(i, j) }
                add(this, j, i)
            }
            c
        } }
    }

    fun updateCells(gameBoard: Array<Array<Boolean>>) {
        for (i in gameBoard.indices) {
            for (j in gameBoard[i].indices) {
                updateSingleCell(i, j)
            }
        }
    }

    fun updateSingleCell(x: Int, y: Int) {
        cells[x][y].update(controller.gameBoard.board[x][y], gridOn)
    }
}
